### PR TITLE
API updates per review feedback

### PR DESCRIPTION
**Pull request by twsouthwick**
https://github.com/dotnet/aspnetcore/issues/42091

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs [hunk 1]
```diff
@@ -50,27 +51,7 @@ public static ISystemWebAdapterBuilder AddRemoteAppClient(this ISystemWebAdapter
...
-    }
-
-    public static ISystemWebAdapterRemoteClientAppBuilder Configure(this ISystemWebAdapterRemoteClientAppBuilder builder, Action<RemoteAppClientOptions> configure)
-    {
-        if (builder is null)
-        {
-            throw new ArgumentNullException(nameof(builder));
-        }
-
-        if (configure is null)
```
Comment (mjrousos) [inline src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs:65]: Should probably keep this error check

File: src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerExtensions.cs [hunk 2]
```diff
@@ -25,29 +25,10 @@ public static ISystemWebAdapterBuilder AddRemoteAppServer(this ISystemWebAdapter
...
-    }
-
-    public static ISystemWebAdapterRemoteServerAppBuilder Configure(this ISystemWebAdapterRemoteServerAppBuilder builder, Action<RemoteAppServerOptions> configure)
-    {
-        if (builder is null)
-        {
-            throw new ArgumentNullException(nameof(builder));
-        }
-
-        if (configure is null)
```
Comment (mjrousos) [inline src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerExtensions.cs:42]: Again, probably want to keep the null check

**Review by mjrousos — APPROVED** [review]
LGTM. A few minor comments inline.

**Review by Tratcher — APPROVED** [review]

File: src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs (hunk 1 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs:65]: The null check is still there - this whole method was removed :)

File: src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerExtensions.cs (hunk 2 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters.FrameworkServices/RemoteAppServerExtensions.cs:42]: same - the method was removed. the null check is still in the existing method

File: src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs (hunk 1 above)
Comment (mjrousos) [reply src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/RemoteAppClientExtensions.cs:65]: Ah, right. 👍

**Pull request merged**